scripts/utils.js

```diff
@@ -76,26 +76,51 @@ const getWeightedSignersSet = (signerAddresses, weights, threshold) => {
     return defaultAbiCoder.encode(['address[]', 'uint256[]', 'uint256'], [sortedAddresses, sortedWeights, threshold]);
 };
 
-const getWeightedSignersProof = async (data, accounts, weights, threshold, signers) => {
-    const signersWithWeights = getAddresses(accounts).map((address, i) => ({ address, weight: weights[i] }));
-    const sortedSignersWithWeights = sortBy(signersWithWeights, (signer) => signer.address.toLowerCase());
-    const sortedAddresses = sortedSignersWithWeights.map(({ address }) => address);
-    const sortedWeights = sortedSignersWithWeights.map(({ weight }) => weight);
-
-    const hash = arrayify(keccak256(data));
-    const signatures = await Promise.all(
-        sortBy(signers, (wallet) => wallet.address.toLowerCase()).map((wallet) => wallet.signMessage(hash)),
+const encodeWeightedSigners = (weightedSigners) => {
+    return defaultAbiCoder.encode(
+        ['tuple(tuple(address signer,uint128 weight)[] signers,uint128 threshold,bytes32 nonce) signers'],
```

Comment:
why do you have `signers` there twice? maybe just keep it types and not var names?

Reply:
Var names are needed to allow using objects for structs, otherwise they need to be encoded as nested arrays

Reply:
Don't need the outermost name, will remove that

Reply:
I though this just works
```
'tuple(tuple(address, uint128)[], uint128, bytes32)'
```